Add guide for customizing your storefront

gsmendoza opened the pull request:
The guide is updated to demonstrate how to customize the frontend. It covers

* Editing a view
* Running the frontend specs
* Adding an image
* Customizing a stylesheet

kennyadsl commented on docs/customization/customizing-your-storefront.mdx:
```diff
@@ -0,0 +1,170 @@
+
+## Installing the frontend
+
+With Solidus 3.2, installing Solidus would automatically install[SolidusStarterFrontend][], which is
```
Missing link for SolidusStarterFrontend? Or this is just a syntax I don't know?

gsmendoza replied:
It's an... "implicit link name shortcut" :p From https://daringfireball.net/projects/markdown/syntax#link:

![image](https://user-images.githubusercontent.com/61476/184312614-f00275db-6ed2-4515-b623-d4ac1025bc41.png)

I should fix that space between "install"  and the link, though :)

aldesantis replied:
@gsmendoza FYI this doesn't seem to be working in the [preview](https://2c3489d1.edgeguides.pages.dev/customization/customizing-your-storefront), so it probably needs to be explicit.

gsmendoza replied:
@aldesantis I forgot to add the link for SolidusFrontend :p Fixing it...

kennyadsl commented on docs/customization/customizing-your-storefront.mdx:
````diff
@@ -0,0 +1,170 @@
+</div>
+```
+
+If we search for views with the CSS class `"hero__container"` in your Solidus app, we find
````
Love that you mentioned the search, it puts readers in the mindset they will need to adopt for the rest of the changes. 👍

kennyadsl commented on docs/customization/customizing-your-storefront.mdx:
```diff
@@ -0,0 +1,170 @@
+## Installing the frontend
+
+With Solidus 3.2, installing Solidus would automatically install[SolidusStarterFrontend][], which is
+the new frontend solution for Solidus. As a Rails application template, it copies all the Solidus
```
I'd move this reference to 3.2 in an info box somewhere in the page. Something like "Before 3.2, we used to do XYZ, here's the link to the legacy solution, etc etc". For the majority of the new users, this information will just be confusing and they should see a clear main way of doing things without distractions. WDYT?

gsmendoza replied:
@kennyadsl I'll try this out:

* Merge Intro and "Installing the frontend" section.
* Move Solidus installation background to info box.

TODO

kennyadsl commented on docs/customization/customizing-your-storefront.mdx:
```diff
@@ -0,0 +1,170 @@
+
+Like the hero partial from earlier, this stylesheet was copied to the Rails app when SolidusStarterFrontend was installed on the app.
+
+SolidusStarterFrontend stylesheets are written with [Sass][].
```
This looks a bit off here. Can we try to add this information in the previous paragraph? Something like:

> Like the hero partial from earlier, this stylesheet (written with [Sass][]) was copied to the Rails app when SolidusStarterFrontend was installed on the app.

gsmendoza replied:
I changed it to "Like the hero partial from earlier, this [Sass][] sstylesheet was copied to the Rails app when SolidusStarterFrontend was installed on the app."

kennyadsl commented on docs/customization/customizing-your-storefront.mdx:
```diff
@@ -0,0 +1,170 @@
+## Summary
+
+Since Solidus copies all of the starter frontend code to your app, modifying frontend views and
+assets is straightforward. You're also provided an initial test suite to test drive your frontend
```
I'd avoid the term `straightforward`. I can see how people without Rails experience can feel overwhelmed at this point and maybe it's not ideal for them to see that they should have understood it at this point. What if we rephrase into:  

> modifying frontend views and assets is similar to any other Rails application.

kennyadsl reviewed (commented):
Nice work, I left some initial comments, but in general I think it's a good start. I'm curious to see if this is what @aldesantis had in mind.

waiting-for-dev commented on docs/customization/customizing-your-storefront.mdx:
````diff
@@ -0,0 +1,170 @@
+
+```html
+<div class="hero__container">
+  <%= image_tag('hero-image') %>
````
What we've been doing in other places when modifying code is adding the diff for it ([example](https://github.com/solidusio/edgeguides/blob/338608abf9f0101c0717f5acb121acb00e717f03/docs/customization/customizing-the-backend.mdx?plain=1#L41)). WDYT?

gsmendoza replied:
Sounds good. I'll update the code using diff :+1:

waiting-for-dev commented on docs/customization/customizing-your-storefront.mdx:
````diff
@@ -0,0 +1,170 @@
+
+```
+
+When you installed SolidusStarterFrontend to your app, you also received some some specs for testing
````
```suggestion
When you installed SolidusStarterFrontend to your app, you also received some specs for testing
```

waiting-for-dev commented on docs/customization/customizing-your-storefront.mdx:
```diff
@@ -0,0 +1,170 @@
+When you installed SolidusStarterFrontend to your app, you also received some some specs for testing
+the frontend. These specs are part of the SolidusStarterFrontend test suite itself, that is, they
+are used in testing the SolidusStarterFrontend project. By copying these specs to your app, you get
+a good starting test suite for testing your application.
```
```suggestion
a good starting point for testing your application.
```

A lot of "test" or variations in the sentence. Maybe we can get rid of this one :slightly_smiling_face:

waiting-for-dev commented on docs/customization/customizing-your-storefront.mdx:
````diff
@@ -0,0 +1,170 @@
+
+```scss
+.hero {
+  // ...
````
Same that before. Does a diff look nice?

waiting-for-dev reviewed (approved):
I like it, very well structured! Thanks, @gsmendoza, I only left minor suggestions and a typo.

gsmendoza commented:
Hi @kennyadsl , @waiting-for-dev ! I updated the PR based on your feedback. I appreciate if you can take a second look. Here's the preview page: https://gsmendoza-eng-138-write-stor.edgeguides.pages.dev/customization/customizing-your-storefront.

aldesantis reviewed (commented):
@gsmendoza other than a broken link, this looks good to me.

gsmendoza commented:
@aldesantis I fixed the broken link and squashed the commits. For some reasons, the PR no longer generated a preview of the site. I'm not sure why. Here's a screenshot of the page:

![image](https://user-images.githubusercontent.com/61476/185293546-a2283a64-035f-4a7b-8346-7d5938ecbf11.png)

aldesantis commented:
@gsmendoza looking good!
